src/ServiceControl/Connection/PlatformConnectionBuilder.cs

```diff
@@ -0,0 +1,26 @@
+﻿namespace ServiceControl.Connection
+{
+    using System.Linq;
+    using System.Threading.Tasks;
+
+    class PlatformConnectionBuilder : IPlatformConnectionBuilder
+    {
+        readonly IProvidePlatformConnectionDetails[] platformConnectionProviders;
+
+        public PlatformConnectionBuilder(IProvidePlatformConnectionDetails[] platformConnectionProviders)
+            => this.platformConnectionProviders = platformConnectionProviders;
+
+        public async Task<PlatformConnectionDetails> BuildPlatformConnection()
+        {
+            var connectionDetails = new PlatformConnectionDetails();
+
+            // TODO: Handle errors in providers (including timeouts?)
```

Comment:
This is [done already](https://github.com/Particular/ServiceControl/pull/2674/files#diff-8b54e2ea88c21bb9db165c40c2a65eb0f58d8fa36322c86da684a211522214bcR33) I think.